metric_learn/constraints.py

```diff
@@ -0,0 +1,63 @@
+""" Helper class that can generate different types of constraints from supervised data labels."""
+
+import random
+import numpy as np
+from six.moves import xrange
+
+class Constraints(object):
```

Comment:
A class containing only static methods isn't all that useful. Either:

 1. Ditch the class and access the functions from the module namespace directly
 2. Make a stateful constraint-manager class, which handles some internal representation of constraints and has methods for emitting them in various formats. This class would then have a single method for initializing from supervised labels.

Option 2 would be ideal, but I'm not sure how feasible it would be. It might also be outside the scope of this PR, and could be added later.